### PR TITLE
CHANGE (CodeAnalyzer): @W-12439992@: Adds a new parameter to control path expansion limits.

### PR DESCRIPTION
**Pull request by rmohan20**
1. Creating a new parameter (`--pathexplimit`) to set path expansion limit through `scanner:run:dfa` invocation
2. Additional logs to check what path expansion limits have been set

### CODE REVIEW

File: messages/run-dfa.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ module.exports = {
 	"commandDescriptionLong": `Scans codebase with all DFA rules by default.
 	Specify the format of output and print results directly or as contents of a file that you provide with --outfile flag.`,
 	"flags": {
+		"pathexplimitDescription": "specify upperlimit to which Graph Engine can handle code complexity. Alternatively, set value using environment variable `SFGE_PATH_EXPANSION_LIMIT`",
+		"pathexplimitDescriptionLong": "Specifies an upperlimit upto which Graph Engine would handle code complexity before failing fast. Set value to -1 to remove any upper limit. Inherits value from SFGE_PATH_EXPANSION_LIMIT env-var if set. Default value is derived from jvm heap space allocation.",
```
Comment (rmohan20) [inline messages/run-dfa.js:7]: @teresa-allen-sfdc Please let me know your thoughts about this help text.

File: sfge/src/main/java/com/salesforce/graph/ops/expander/PathExpansionRegistry.java [hunk 2]
```diff
@@ -73,6 +73,19 @@ private static class ApexPathExpanderRegistryData extends RegistryData<ApexPathE
         ApexPathExpanderRegistryData() {
             pathExpansionLimit = SfgeConfigProvider.get().getPathExpansionLimit();
             isPathExpansionLimitSet = pathExpansionLimit > 0;
+
+            if (LOGGER.isInfoEnabled()) {
```
Comment (rmohan20) [inline sfge/src/main/java/com/salesforce/graph/ops/expander/PathExpansionRegistry.java:77]: New log line we'll print to surface the limit numbers and what heap space this is based on.

File: sfge/src/main/java/com/salesforce/graph/ops/registry/RegistryDataLimitCalculator.java [hunk 3]
```diff
@@ -48,12 +48,9 @@ int calculateAllowedLimit(long averageItemSize, int minimumItemCountExpected) {
 
         final int allowedLimit = (int) ((heapMaxSize * CAPACITY_LIMIT) / averageItemSize);
 
-        if (LOGGER.isInfoEnabled()) {
```
Comment (rmohan20) [inline sfge/src/main/java/com/salesforce/graph/ops/registry/RegistryDataLimitCalculator.java:51]: This log line never got printed.

File: sfge/src/test/java/com/salesforce/graph/build/MethodPathBuilderTest.java [hunk 4]
```diff
@@ -2743,6 +2744,59 @@ public void testMethodSwitchStatement() {
         MatcherAssert.assertThat(paths, hasSize(equalTo(3)));
     }
 
+    @Disabled // TODO: Allow Switch statements to lookup array load values.
```
Comment (rmohan20) [inline sfge/src/test/java/com/salesforce/graph/build/MethodPathBuilderTest.java:2747]: Unrelated tests. I added them to repro a user issue.

File: src/commands/scanner/run/dfa.ts [hunk 5]
```diff
@@ -68,6 +68,11 @@ export default class Dfa extends ScannerRunCommand {
 			description: messages.getMessage('flags.sfgejvmargsDescription'),
 			longDescription: messages.getMessage('flags.sfgejvmargsDescriptionLong'),
 			env: 'SFGE_JVM_ARGS'
+		}),
+		'pathexplimit': flags.integer({
```
Comment (rmohan20) [inline src/commands/scanner/run/dfa.ts:72]: New parameter

Comment (jfeingold35) [reply src/commands/scanner/run/dfa.ts:72]: The convention is to use kebab-casing. Could we call this `path-exp-limit`?

**Review by jfeingold35 — APPROVED** [review]

**Pull request merged**